File: src/gui/macos/Alerts.m
#import "QuarkMac.h"

int QuarkRunMessageAlert(NSString *kind, NSString *title, NSString *body) {
    NSAlert *alert = [[NSAlert alloc] init];
    alert.messageText = title;
    alert.informativeText = body;
    alert.alertStyle = [kind isEqualToString:@"error"] ? NSAlertStyleCritical : NSAlertStyleInformational;
    [alert addButtonWithTitle:@"OK"];
    [NSApp activateIgnoringOtherApps:YES];
    [alert runModal];
    return 0;
}

void QuarkShowModalError(NSString *message, NSWindow *window) {
    NSAlert *alert = [[NSAlert alloc] init];
    alert.messageText = @"Quark Downloader";
    alert.informativeText = message;
    alert.alertStyle = NSAlertStyleCritical;
    [alert addButtonWithTitle:@"OK"];
    if (window != nil) {
        [alert beginSheetModalForWindow:window completionHandler:nil];
    } else {
        [alert runModal];
    }
}

File: src/gui/macos/Alerts.swift
import AppKit

func runMessageAlert(kind: String, title: String, body: String) -> Never {
    let alert = NSAlert()
    alert.messageText = title
    alert.informativeText = body
    alert.alertStyle = kind == "error" ? .critical : .informational
    alert.addButton(withTitle: "OK")
    NSApp.activate(ignoringOtherApps: true)
    alert.runModal()
    exit(0)
}

func showModalError(_ message: String, window: NSWindow?) {
    let alert = NSAlert()
    alert.messageText = "Quark Downloader"
    alert.informativeText = message
    alert.alertStyle = .critical
    alert.addButton(withTitle: "OK")
    if let window {
        alert.beginSheetModal(for: window)
    } else {
        alert.runModal()
    }
}
